airbyte-integrations/connectors/source-netsuite/Dockerfile

```diff
@@ -32,6 +36,10 @@ COPY main.py ./
 
 COPY source_netsuite ./source_netsuite
 
+# Set environment variables in the final image
+ENV HTTP_PROXY $IMAGE_HTTP_PROXY
```

Comment:
suggestion: If you are going to get the proxy from outside, it is probably better to take it as a configuration option rather than a baked in setting. You can change the configuration quickly but you cannot re-build and deploy a connector quickly.

Reply:
You're right. We need to provide PROXY as an env parameter. The problem is we deploy Airbyte using Helm charts so helm charts are in the charge of passing env variables to pods. However, there is a bug for this and we cannot pass env variable to read and write pods. We can provide only an image to helm chart so this image should contain the correct PROXY variables.

As you said, we need to re-build and re-deploy if PROXY changes. This solution is a temporary solution. We have an [open issue](https://github.com/airbytehq/airbyte/issues/38158) for it and after the resolution we will continue with the actual solution.

Reply:
You can set the environment variable from inside the connector code if you pass is through the config. [Example](https://github.com/canonical/airbyte/pull/1/commits/d0df04926e4d96c9fdd45a878802fa85fd05df45#diff-6448bb711c7f83bc12d82a258467e7c35d68921de6af7022eb4f0466683f2cceR98).

Reply:
Got your point. The disadvantages of passing variables inside the code are:
* We need to change the source code of the connector but we will also rollback it once the permanent solution is ready. Also, there is not just a single place to change in code, we need to handle everywhere making an HTTP request.
* We may skip a part of code which sends an HTTP request to an unknown address. For instance, maybe the connector will call the route /`api/check `to check the health of the Netsuite Server and this may not be a part of connector code but rather a part of an dependency.  Handling this each case is not easy so providing PROXY parameter as environment variable is a better option.

Reply:
I agree with Mert to an extent, have you tried adding the proxy variables as configs to the connector and running the sync to test it?

Reply:
Yeap tried but faced Proxy problems related to dependant libraries. There is not just a single point of entry for calling Netsuite API. This is the second of the disadvantages.  My first point may not be as critical as first one but it also worth considering because in the rollback scenario we need to retest our connector because we will make a significant code change.